SlevomatCodingStandard/Sniffs/Attributes/AttributesOrderSniff.php

```diff
@@ -76,34 +85,70 @@ public function process(File $phpcsFile, $attributeOpenerPointer): void
 
 		} while (true);
 
+		$expectedOrder = [];
 		$actualOrder = [];
 
-		foreach ($attributesGroups as $attributesGroupNo => $attributesGroup) {
-			$attributeName = $this->normalizeAttributeName($attributesGroup[0]->getName());
-
-			foreach ($this->order as $orderPosition => $attributeNameOnPosition) {
-				if (
-					$attributeName === $attributeNameOnPosition
-					|| (
-						substr($attributeNameOnPosition, -1) === '\\'
-						&& strpos($attributeName, $attributeNameOnPosition) === 0
-					)
-					|| (
-						substr($attributeNameOnPosition, -1) === '*'
-						&& strpos($attributeName, substr($attributeNameOnPosition, 0, -1)) === 0
-					)
-				) {
-					$actualOrder[$attributesGroupNo] = $orderPosition;
-					continue 2;
+		if ($this->orderAlphabetically) {
+			$actualOrder = $attributesGroups;
+
+			uasort($attributesGroups, static function (array $attributesGroup1, array $attributesGroup2): int {
+				$content1 = $attributesGroup1[0]->getContent();
+				$content2 = $attributesGroup2[0]->getContent();
+
+				if ($content1 === null && $content2 === null) {
+					return 0;
 				}
-			}
 
-			// Unknown order - add to the end
-			$actualOrder[$attributesGroupNo] = 999;
+				if ($content1 === null) {
+					return -1;
+				}
+
+				if ($content2 === null) {
+					return 1;
+				}
+
+				return strnatcmp($content1, $content2);
+			});
+
+			uasort($attributesGroups, static function (array $attributesGroup1, array $attributesGroup2): int {
+				return strnatcmp($attributesGroup1[0]->getName(), $attributesGroup2[0]->getName());
+			});
+
+			$expectedOrder = $attributesGroups;
+
+			if ($expectedOrder === $actualOrder) {
```

Comment:
Still duplicate code.

And `if ($this->orderAlphabetically === false) {` should be `else`.

Reply:
Refactored.